packages/google_mobile_maps/lib/google_mobile_maps.dart

```diff
@@ -0,0 +1,173 @@
+/// into the Flutter application.
+class GoogleMapsController {
+  /// An ID identifying the GoogleMaps instance, once created.
+  final Future<int> id;
+
+  GoogleMapsController(this.id);
+
+  /// Initiate a camera move to the specified [location] and [zoom] level.
+  Future<void> moveCamera(Location location, Zoom zoom) async {
+    final int id = await this.id;
+    await _channel.invokeMethod('moveCamera', <String, dynamic>{
+      'id': id,
+      'location': location._toJson(),
+      'zoom': zoom._toJson()
+    });
+  }
+}
+
+/// Controller for a GoogleMaps instance that is integrated as a
+/// platform overlay.
+///
+/// *Warning*: Platform overlays cannot be freely composed with
+/// over widgets. See [PlatformOverlayController] for caveats and
+/// limitations.
+class GoogleMapsOverlayController {
+  GoogleMapsOverlayController._(this.mapsController, this.overlayController);
+
+  /// Creates a controller for a GoogleMaps of the specified size in
+  /// logical pixels.
+  factory GoogleMapsOverlayController.fromSize(double width, double height) {
+    final _GoogleMapsPlatformOverlayController overlayController =
+        new _GoogleMapsPlatformOverlayController(width, height);
+    final GoogleMapsController mapsController =
+        new GoogleMapsController(overlayController.mapId);
+    return new GoogleMapsOverlayController._(mapsController, overlayController);
+  }
+
+  /// The controller of the GoogleMaps instance.
+  final GoogleMapsController mapsController;
+
+  /// The controller of the platform overlay.
+  final PlatformOverlayController overlayController;
+
+  void dispose() {
+    overlayController.dispose();
+  }
+}
+
+class _GoogleMapsPlatformOverlayController extends PlatformOverlayController {
+  Completer<int> _textureId = new Completer<int>();
+
+  _GoogleMapsPlatformOverlayController(double width, double height)
+      : super(width, height);
+
+  Future<int> get mapId => _textureId.future;
+
+  @override
+  Future<int> createOverlay(Size physicalSize) {
+    _textureId.complete(_channel.invokeMethod('createMap', <String, dynamic>{
+      'width': physicalSize.width,
+      'height': physicalSize.height,
+    }).then<int>((dynamic value) => value));
+    return _textureId.future;
+  }
+
+  @override
+  Future<void> showOverlay(Offset physicalOffset) async {
+    final int id = await _textureId.future;
+    _channel.invokeMethod('showMapOverlay', <String, dynamic>{
+      'id': id,
+      'x': physicalOffset.dx,
+      'y': physicalOffset.dy,
+    });
+  }
+
+  @override
+  Future<void> hideOverlay() async {
+    final int id = await _textureId.future;
+    _channel.invokeMethod('hideMapOverlay', <String, dynamic>{
+      'id': id,
+    });
+  }
+
+  @override
+  Future<void> disposeOverlay() async {
+    final int id = await _textureId.future;
+    _channel.invokeMethod('disposeMap', <String, dynamic>{
+      'id': id,
+    });
+  }
+}
+
+/// A Widget that is fronted by a GoogleMaps platform overlay.
```

Comment:
'fronted'?

Reply:
Changed to "covered by". But dictionary says

> provide (something) with a front or facing of a particular type or material.
"a metal box fronted by an alloy panel"